pkg/builder/builder.go

```diff
@@ -121,48 +125,36 @@ func (payloadConfig *PayloadConfig) GeneratePayload(filename string, payload []b
 	var buildmode string
 	isDLL := false
 
-	switch strings.ToLower(payloadConfig.Payload.Type) {
-	case "exe":
-		extension = "exe"
-	case "dll":
-		extension = "dll"
-		buildmode = "c-shared"
-		isDLL = true
-	case "cpl":
-		extension = "cpl"
-		buildmode = "c-shared"
-		isDLL = true
-	case "xll":
-		extension = "xll"
-		buildmode = "c-shared"
-		isDLL = true
-	case "pie":
-		extension = "exe"
-		buildmode = "pie"
-	default:
-		return "", nil, fmt.Errorf("Type must be exe, dll, cpl, xll, or pie.")
-	}
+	// create compiler object
+	thisCompiler := compiler.GetEmptyCompiler(payloadConfig.Payload.Lang)
+
+	// get build info for specific compiler, verify config is supported (dll, exe etc.)
+	extension, buildmode, err = thisCompiler.IsTypeSupported(payloadConfig.Payload.Type)
```

Comment:
This is cool! I think we should add a "IsCompilerInstalled", to check if the compiler is installed on the machine

Reply:
good idea, i hard set the path for mingw compilers too, so that should be fixed as well